miden-lib/src/notes/mod.rs

```diff
@@ -50,6 +58,31 @@ pub fn create_note(
             ProgramAst::from_bytes(p2idr_bytes).map_err(NoteError::NoteDeserializationError)?,
             vec![target.into(), recall_height.into(), ZERO, ZERO],
         ),
+        Script::ASWAP {
+            asset,
+            serial_num,
+            tag,
+        } => {
+            let recipient = build_p2id_recipient(sender, serial_num)?;
+            let asset_word: Word = asset.into();
+            (
+                ProgramAst::from_bytes(aswap_bytes).map_err(NoteError::NoteDeserializationError)?,
+                vec![
+                    recipient[0],
+                    recipient[1],
+                    recipient[2],
+                    recipient[3],
+                    asset_word[0],
+                    asset_word[1],
+                    asset_word[2],
+                    asset_word[3],
+                    tag,
+                    ZERO,
+                    ZERO,
+                    ZERO,
+                ],
```

Comment:
Here, `tag` could be set to `sender`. This way, we don't need to provide the tag via `ASWAP` enum.